File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/HashCode.java
package cesiumlanguagewriter;


import agi.foundation.compatibility.*;
import javax.annotation.Nonnull;
import javax.annotation.Nullable;

/**
 * Contains static methods for working with hash codes.
 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public final class HashCode {
    private HashCode() {}

    /**
    * Combines an arbitrary number of hash codes.
    * @param hashes The hash codes.
    * @return The combined hash code.
    */
    public static int combine(@Nonnull int... hashes) {
        int result = 17;
        for (final int hash : hashes) {
            result = result * 31 + hash;
        }
        return result;
    }

    /**
    * Gets a hash code for a specified object, or zero if the object is {@code null}.
    * @param o The object for which to get the hash code.
    * @return The hash code of the object, or zero if the object is {@code null}.
    */
    public static int getHashCode(@Nullable Object o) {
        return o == null ? 0 : o.hashCode();
    }
}

File: DotNet/CesiumLanguageWriter/HashCode.cs
using JetBrains.Annotations;

namespace CesiumLanguageWriter
{
    /// <summary>
    /// Contains static methods for working with hash codes.
    /// </summary>
    public static class HashCode
    {
        /// <summary>
        /// Combines an arbitrary number of hash codes.
        /// </summary>
        /// <param name="hashes">The hash codes.</param>
        /// <returns>The combined hash code.</returns>
        public static int Combine([NotNull] params int[] hashes)
        {
            int result = 17;
            foreach (int hash in hashes)
            {
                result = result * 31 + hash;
            }
            return result;
        }

        /// <summary>
        /// Gets a hash code for a specified object, or zero if the object is <see langword="null"/>.
        /// </summary>
        /// <param name="o">The object for which to get the hash code.</param>
        /// <returns>The hash code of the object, or zero if the object is <see langword="null"/>.</returns>
        public static int GetHashCode([CanBeNull] object o)
        {
            return o == null ? 0 : o.GetHashCode();
        }
    }
}